init.c

```diff
@@ -662,7 +662,7 @@ static int fixup_options(struct thread_data *td)
 	    !o->norandommap) {
 		log_err("fio: Any use of blockalign= turns off randommap\n");
 		o->norandommap = 1;
-		ret = warnings_fatal;
+		ret += warnings_fatal;
```

Comment:
This is better, just fold this with the previous patch and make your new user do the same.